spyder/plugins/onlinehelp/plugin.py

```diff
@@ -66,6 +66,13 @@ def register(self):
         widget.load_history(self.load_history())
         widget.sig_load_finished.connect(self.sig_load_finished)
 
+    def on_mainwindow_visible(self):

```

Comment:
I don't like this solution too much because starting the Pydoc server doesn't depend on the main window being visible. It depends on the plugin being visible. So, although this works, I find it a workaround that makes things confusing (I'm just thinking on the maintainability of this code in the future).

So I'd prefer to use the previous solution, which in the new API would translate (I think) to making `widget.server_enabled = True` when `widget.sig_toggle_view_changed` is emitted and then running `widget.initialize()`.

The problem with my proposal seems to be (if I'm not mistaken) that now `widget.sig_toggle_view_changed` is emitted for all plugins at startup, whether they are visible or not. If that's the case, could you explain why?